lib/db.js

```diff
@@ -594,6 +594,30 @@ module.exports = (
     })
   }
 
+  DB.prototype.updateSessionToken = function (sessionToken, geo) {
+    const { id, uid } = sessionToken
+
+    log.trace({ op: 'DB.updateSessionToken', id, uid })
+
+    return this.touchSessionToken(sessionToken, geo)
```

Comment:
Hmm, just wondering: as well as applying the update to Redis, would it also be valid to just delete this token from Redis entirely now, because all of the updated data is in MySQL? And I guess if we were going to do that safely, the delete would need to be moved after the update instead of before. But it would work and also leave us with a little bit less data in Redis, wouldn't it?

Reply:
Or maybe the really safe way would be:

1. update in Redis
2. update in MySQL
3. delete from Redis, catching any error as non-fatal

Reply:
I think this should be valid, but I wonder if the potential space saving is worth the extra step, or if we're in an OK place with redis usage now and don't need to worry about it.

Reply:
Fair enough, although there is also a (very low probability) correctness benefit to it as well fwiw.

We know our Redis-pruning isn't perfect, because only tokens in the expired-but-not-yet-pruned-from-MySQL state are candidates for it. So we still have a (very low) risk of future session token id clashes. Deleting here would further reduce that (already very low) risk.

So I agree maybe it's not worth doing, but just wanted to note the other potential benefit. (and maybe I'll open a follow-up issue for triage if you don't want to do it here)

Reply:
Actually, it just occurred to me we could entirely mitigate the aforementioned risk if we pro-actively deleted from Redis in `createSessionToken`. Not sure why that didn't occur to me before.